worker/metrics/collect/manifold.go

```diff
@@ -151,10 +160,28 @@ func newCollect(config ManifoldConfig, context dependency.Context) (*collect, er
 		unitTag: unitTag.String(),
 		paths:   paths,
 	}
-	var listener stopper
-	charmURL, validMetrics, err := readCharm(unitTag, paths)
+	var (
+		listener     stopper
+		charmURL     *corecharm.URL
+		validMetrics map[string]corecharm.Metric
+	)
+	err = retry.Call(retry.CallArgs{
+		Func: func() error {
+			var err error
+			charmURL, validMetrics, err = readCharm(unitTag, paths)
+			return err
+		},
+		IsFatalError: func(err error) bool {
+			return errors.Cause(err) != errCharmDirNotExist
+		},
+		Attempts:    10,
+		Delay:       time.Second,
+		MaxDelay:    time.Duration(30) * time.Second,
```

Comment:
This is 1 2 4 8 16 30 30 30 30 30 delays. Or roughly 3 minutes. I'm not sure that's the right length of time. I guess as long as metric collector is running independently of other hooks so it is only blocking itself from running.